### ISSUE TITLE
Improve updog's anonymous metrics

### ISSUE THREAD

**Issue by webern** [bottlerocket-os/bottlerocket#1001]
## What I'd like:

Once we land #1000, `updog`'s reporting of anonymous metrics should be improved to:

 * include additional fields, in-line with those of ~~`healthdog`~~ `metricdog`
 * use common code, or otherwise make use of ~~`healthdog`'s~~ `metricdog`'s reporting mechanism

At a minimum, `updog` should report some additional fields, as follows:

### updog check-update

* sender: updog ( ← new )
* event: check-update ( ← new )
* version: 0.5.3
* variant: aws-k8s-1.17 ( ← new )
* arch: x86_64 ( ← new )
* region: us-west-2 ( ← new )
* seed: 1292
* version-lock: latest ( ← new )
* ignore-waves: false ( ← new ): resolved value (setting or override at command line)

### updog update

* sender: updog ( ← new )
* event: update ( ← new )
* version: 0.5.3
* variant: aws-k8s-1.17 ( ← new )
* arch: x86_64 ( ← new )
* region: us-west-2 ( ← new )
* seed: 1292
* version-lock: latest ( ← new )
* ignore-waves: false ( ← new ): resolved value (setting or override at command line)
* target: 0.6.0
* migrations: migrationabz.lz4,migrationxyz.lz4

## Any alternatives you've considered:

We've discussed two designs so far without picking one.
 * `healthdog` provides a `lib.rs` and `updog` uses that library to make the reports.
 * `updog` writes its information to a file, and `healthdog` reads information from that file at shutdown, on a schedule, or otherwise with some appropriate timing.

**Comment by stmcginnis** [comment]
@webern is there more that needs to be done beyond what https://github.com/bottlerocket-os/bottlerocket/issues/1000 covered?

**Comment by webern** [comment]
> is there more that needs to be done beyond what https://github.com/bottlerocket-os/bottlerocket/issues/1000 covered?

Yes, there is a lot more information that would be useful from Bottlerocket hosts. This issue, as written, seems to be about reporting updog downloads. It's probably something that we want but is not currently prioritized.

**Comment by bcressey** [comment]
At this point I'm more inclined to delete the metrics code altogether than to expand its use.
